File: WorkFlows/Chapter02/VBFirstSequentialWFConsoleApplication/Workflow1.vb
Imports System
Imports System.ComponentModel
Imports System.ComponentModel.Design
Imports System.Collections
Imports System.Drawing
Imports System.Workflow.ComponentModel.Compiler
Imports System.Workflow.ComponentModel.Serialization
Imports System.Workflow.ComponentModel
Imports System.Workflow.ComponentModel.Design
Imports System.Workflow.Runtime
Imports System.Workflow.Activities
Imports System.Workflow.Activities.Rules

'NOTE: When changing the namespace; please update XmlnsDefinitionAttribute in AssemblyInfo.vb
Partial Public Class Workflow1
    Inherits SequentialWorkflowActivity
    Private InputValue1 As Integer
    Private InputValue2 As Integer
    Private OutputResult As Integer
    Public WriteOnly Property Input1() As Integer
        Set(ByVal value As Integer)
            InputValue1 = value
        End Set
    End Property
    Public WriteOnly Property Input2() As Integer
        Set(ByVal value As Integer)
            InputValue2 = value
        End Set
    End Property
    Public ReadOnly Property OutputValue() As Integer
        Get
            Return OutputResult
        End Get
    End Property
    Public Sub New()
        MyBase.New()
        InitializeComponent()
    End Sub

    Private Sub Step1_ExecuteCode(ByVal sender As System.Object, ByVal e As System.EventArgs)
        OutputResult = InputValue1 + InputValue2
        MsgBox("Step1")
    End Sub

    Private Sub Step2_ExecuteCode(ByVal sender As System.Object, ByVal e As System.EventArgs)
        MsgBox("Step2")
    End Sub

    
End Class

File: WorkFlows/Chapter02/CFirstSequentialWFConsoleApplication/Workflow1.cs
using System;
using System.ComponentModel;
using System.ComponentModel.Design;
using System.Collections;
using System.Drawing;
using System.Workflow.ComponentModel.Compiler;
using System.Workflow.ComponentModel.Serialization;
using System.Workflow.ComponentModel;
using System.Workflow.ComponentModel.Design;
using System.Workflow.Runtime;
using System.Workflow.Activities;
using System.Workflow.Activities.Rules;

namespace CFirstSequentialWFConsoleApplication
{
	public sealed partial class Workflow1: SequentialWorkflowActivity
	{
        private int InputValue1;
        private int InputValue2;
        private int OutputResult;

        public int Input1
        {
            set {
                InputValue1 = value;
            }
        }

        public int Input2
        {
            set
            {
                InputValue2 = value;
            }
        }

        public int OutputValue
        {
            get
            {
                return OutputResult;
            }
        }

		public Workflow1()
		{
			InitializeComponent();
		}

        private void Step1_ExecuteCode(object sender, EventArgs e)
        {
            OutputResult = InputValue1 + InputValue2;
            Console.WriteLine("Step1");
        }

        private void Step2_ExecuteCode(object sender, EventArgs e)
        {
            Console.WriteLine("Step2");
        }

       

	}

}
